src/book.js

```diff
@@ -299,7 +299,11 @@ export class Book {
     //  create a directory named after the input pdf and fill it with
     //  the signatures
     this.zip = new JSZip();
-    var origFileName = this.inputpdf.replace(/\s|,|\.pdf/, '');
+    var origFileName = this.inputpdf;
+    origFileName = origFileName.replace(/\s|-+|,/g, '_');
+    origFileName = origFileName.replace(/_+/g, '_');
```

Comment:
`this.filename = origFileName.replace(/[-\s,_]+/g, '_').replace(/_*\.pdf/g, '').toLowerCase();` should replace lines 303-307

Reply:
I chained the calls but the regex's need to stay as they are or they don't work correctly.

Reply:
Also added comments to the regex's

Reply:
to your point I also think what you should wrote, but I tested it and it doesn't seem to function the same and I blame javascript for being haunted.

Reply:
it should be doing the same thing - do you have s ample string where the outputs are different? Also I think you have a typo ('time' instead of 'trim') in the last of the comments.

Reply:
comment changed

Reply:
using the code you provided the zip file generated is 'Kitty Cat Kill Sat - Argus.pdf_bookbinder.zip' 

```
    //  create a directory named after the input pdf and fill it with
    //  the signatures
    this.zip = new JSZip();
    var origFileName = this.inputpdf;
    this.filename = origFileName.replace(/[-\s,_]+/g, '_').replace(/_*\.pdf/g, '').toLowerCase();
```

🤷